packages/flutter/lib/src/material/tabs.dart

```diff
@@ -562,8 +562,12 @@ class TabBarSelectionState<T> extends State<TabBarSelection<T>> with SingleTicke
   @override
   void didUpdateConfig(TabBarSelection<T> oldConfig) {
     super.didUpdateConfig(oldConfig);
+    _previousValue = _value;
+    _value = config.value ?? values.first;
```

Comment:
I think it would be reasonable if the widget's value was called initialValue and TabBarSelectionState's public states was read-only.  I just tried doing as much but unfortunately that revealed deeper problems.

I think the overall API needs some thought. Using it can be clumsy the internal logic for reconfiguring the TabBar's children when the user swipes vs taps is pretty hair-raising.  Likewise for the TabBar code that tries to stay in sync with changes to its TabBarSelection ancestor. I'd rather tackle a bigger revision in a separate PR if that's OK with you.